skyportal/handlers/api/internal/profile.py

```diff
@@ -246,3 +246,70 @@ def patch(self, user_id=None):
             if "sourceCounts" in preferences:
                 self.push(action="skyportal/FETCH_SOURCE_COUNTS")
             return self.success(action="skyportal/FETCH_USER_PROFILE")
+
+    @auth_or_token
+    def post(self, user_id=None):
+        """
+        ---
+        description: add gcn preference
+        requestBody:
+          content:
+            application/json:
+              type: object
+                properties:
+                  preferences:
+                    schema: AddUserPreferencesRequestJSON
+                    description: JSON describing updates to user preferences dict
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        data = self.get_json()
+
+        with self.Session() as session:
+            if 'user_id' in data:
+                user_id = data['user_id']
+                user = session.scalars(
+                    User.select(session.user_or_token).where(User.id == user_id)
+                ).first()
+            else:
+                user = self.associated_user_object
+                user_id = user.id
+
+            token_acls = set(data['acls'])
+            if not all([acl_id in user.permissions for acl_id in token_acls]):
+                return self.error(
+                    "User has attempted to grant token ACLs they do not have "
+                    "access to. Please try again."
+                )
+
+            preferences = data.get("preferences", {})
+
+            # Do not save blank fields (empty strings)
+            for k, v in preferences.items():
+                if isinstance(v, dict):
+                    preferences[k] = {key: val for key, val in v.items() if val != ""}
+            user_prefs = deepcopy(user.preferences)
+
+            if not user_prefs:
+                user_prefs = {}
+            for key, value in preferences.items():
+                if key not in user_prefs:
+                    user_prefs[key] = value
+            try:
+                session.add(user_prefs)
```

Comment:
There isn't actually a user preferences data model is there or maybe I misunderstand?

Reply:
I thought I need to do a post method because with patch I can't create a new gcn events preference in the database. But maybe it is wrong ? Patch works too ?

Reply:
regarding the patch method, I thought there was user preferences data model

Reply:
I think patch is fine, as it should be the same as the rest I think.

Reply:
ok I will do that so ! thank you